### ISSUE TITLE
Can we make pool.ErrClosed public? Or provide a public method to determine if a given error is it?

### ISSUE THREAD

**Issue by edoger** [redis/go-redis#1288]
Many times, we need to downgrade to another instance after one client instance is closed, but it seems insecure to detect errors through string matching.

**Comment by ferhatelmas** [comment]
related https://github.com/go-redis/redis/issues/179

**Comment by l1ving** [comment]
Nothing?

**Comment by edoger** [comment]
Is there any progress?

**Comment by ndyakov** [comment]
Makes sense. Assigning this to myself and will review the current state if it needs to be addressed.

**Comment by ndyakov** [comment]
@edoger after some investigation I can see the error is already exposed in `error.go`:
https://github.com/redis/go-redis/blob/482a20d9c970c7ba99181290586ca1c9122a5478/error.go#L16

Marking this as done.
